lib/hooks/csrf/index.js

```diff
@@ -77,8 +77,14 @@ module.exports = function(sails) {
           var allowCrossOriginCSRF = sails.config.csrf.origin.split(',').map(trim).indexOf(req.headers.origin) > -1;
 
           if (sails.config.csrf.protectionEnabled) {
-            var connect = require('express/node_modules/connect');
-
+            var connect;
+            try {
+              // Support npm v3 flat dependency installations.
+              connect = require('connect');
```

Comment:
This has more to do with the way npm 3 installs dependencies. With npm 3, a fresh clone of sails (at the current version of express of 3.x), `npm install` will install express' connect dependency adjacent to the express directory. See npm's [3.0.0 release notes](https://github.com/npm/npm/blob/master/CHANGELOG.md#flat-flat-flat) for more details.